src/main/java/coordinateCalculator/Rectangle.java

```diff
@@ -0,0 +1,39 @@
+package coordinateCalculator;
+
+import java.util.*;
+import java.util.function.Function;
+
+public class Rectangle extends AbstractFigure {
+    private static final String MESSAGE_INVALID_RECTANGLE = "입력받은 사각형은 직사각형이 아닙니다.";
+
+    public Rectangle(List<Point> points) {
+        super(points);
+        checkValidRectangle();
+    }
+
+    private void checkValidRectangle() {
+        if (checkValidCoordinates('x') || checkValidCoordinates('y')) {
+            throw new IllegalArgumentException(MESSAGE_INVALID_RECTANGLE);
+        }
+    }
+
+    private boolean checkValidCoordinates(char xOrY) {
+        Map<Character, Boolean> getters = new HashMap<>();
+        getters.put('x', points.stream().map(Point::getX).distinct().count() != 2);
+        getters.put('y', points.stream().map(Point::getY).distinct().count() != 2);
+
+        return getters.get(xOrY);
+    }
```

Comment:
음... 이렇게 빼내는것도 신박하네 ...
```suggestion
    private boolean checkValidCoordinates() {
     points.stream().map(Point::getX).distinct().count() != 2 ||
     points.stream().map(Point::getY).distinct().count() != 2
    }
```
이정도로 생각했었음 ㅋㅋ